types/detect/detect.go

```diff
@@ -22,8 +22,9 @@ type Signature interface {
 }
 
 type SignatureContext struct {
-	Callback SignatureHandler
-	Logger   Logger
+	Callback      SignatureHandler
+	Logger        Logger
+	GetDataSource func(namespace string, id string) (DataSource, bool)
```

Comment:
I thought putting a struct that implements some defined interface.
It is still possible to pass here a functions of an object to change the behavior in the future.
I think it is better to use defined structs rather than unnamed functions as the interface here.

Reply:
What's the interface we need here besides requesting the `DataSource` the signature needs?
As I see it, from the signature's PoV the function _is_ it's interface for requesting the DataSource.

Reply:
So I agree that in Go it is possible to look at this function as an interface.
However, you encourage the usage of hardcoded function instead of more robust approach.
This distinction is not so important, so I won't be persistent on my opinion here. Function is good enough interface.